Generated C++ parser does not allow whitespace to be a token

The following token definition generates bad C++ parser.

`token Foo '\'' ["01x "]* '\'' ;`

The lex file generated is `<YYINITIAL>'( |0|1|x)*'        yylval._string = strdup(yytext); return _SYMB_171;` but it needs to be `<YYINITIAL>'(\ |0|1|x)*'       yylval._string = strdup(yytext); return _SYMB_171;` for it to compile.

The make file reports the following error message

```
flex -Pasl_ -oLexer.C ASL.l
ASL.l:219: unrecognized rule
ASL.l:220: unrecognized rule
make: *** [Lexer.C] Error 1
```

--- a/source/src/BNFC/Backend/C/RegToFlex.hs
+++ b/source/src/BNFC/Backend/C/RegToFlex.hs
@@ -81,4 +81,4 @@ escapeChar c
   | otherwise               = showLitChar c ""
   where
   reserved :: String
-  reserved = "$+-*=<>[](){}!?.,;:^~|&%#/\\$_@\""
+  reserved = " $+-*=<>[](){}!?.,;:^~|&%#/\\$_@\""

--- a/testing/regression-tests/256_Regex/good01.in
+++ b/testing/regression-tests/256_Regex/good01.in
@@ -1,1 +1,1 @@
-[x:4] /\ 'funky+identifier-umlaute:äöüß//\%#FOO and Un]'
+' x10 0' [x:4] /\ 'funky+identifier-umlaute:äöüß//\%#FOO and Un]'

--- a/testing/regression-tests/256_Regex/good01.out
+++ b/testing/regression-tests/256_Regex/good01.out
@@ -3,8 +3,8 @@ Parse Successful!
 
 [Abstract Syntax]
 
-Start (Name "[x:4]") (Name "'funky+identifier-umlaute:\228\246\252\223//\\%#FOO") (Triple "Un]'")
+Start (SpaceInAlts "' x10 0'") (Name "[x:4]") (Name "'funky+identifier-umlaute:\228\246\252\223//\\%#FOO") (Triple "Un]'")
 
 [Linearized tree]
 
-[x:4] /\ 'funky+identifier-umlaute:äöüß//\%#FOO and Un]'
+' x10 0' [x:4] /\ 'funky+identifier-umlaute:äöüß//\%#FOO and Un]'

--- a/testing/regression-tests/256_Regex/test.cf
+++ b/testing/regression-tests/256_Regex/test.cf
@@ -1,6 +1,10 @@
 -- Issue #110: backslash in keyword
 
-Start. Main ::= Name "/\\" Name "and" Triple;
+Start. Main ::= SpaceInAlts Name "/\\" Name "and" Triple;
+
+-- Issue #431: whitespace in character alternatives
+
+token SpaceInAlts '\'' ["01x "]* '\'' ;
 
 -- Issue #319: quotation rule in ANTLR lexer definition
 

